### ISSUE TITLE
drop_connect is effectively dropout here?

### ISSUE THREAD

**Issue by amsword** [zylo117/Yet-Another-EfficientDet-Pytorch#271]
Is drop_connect() here the same as what dropout does? Can we use torch.nn.Dropout instead?

```
def drop_connect(inputs, p, training):
    """ Drop connect. """
    if not training: return inputs
    batch_size = inputs.shape[0]
    keep_prob = 1 - p
    random_tensor = keep_prob
    random_tensor += torch.rand([batch_size, 1, 1, 1], dtype=inputs.dtype, device=inputs.device)
    binary_tensor = torch.floor(random_tensor)
    output = inputs / keep_prob * binary_tensor
    return output
```

https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/f24cbea1a9f94debb7b02285831a1750d113dfdf/efficientnet/utils.py#L85

**Comment by zylo117** [comment]
No, I don't think so.

This is from nn.Dropout2d.

> Randomly zero out entire channels

But drop_connect here randomly one out or zero out entire channels.
